File: Máximo Común Divisor/Python/MCD_Por_Fuerza_Bruta.py
def MCD(a,b):
    if a>=b:
        divisor=a
    else:
        divisor=b
    while divisor>=1:
        if a%divisor==0 and b%divisor==0:
            return divisor;	
        divisor-=1
        
a=int(input("Ingrese número a: "))
b=int(input("Ingrese número b: "))
if a>0 and b>0:
    print("El MCD({:d},{:d})={:d}".format(a,b,MCD(a,b)))
else:
    print("Números inválidos")

File: Máximo Común Divisor/Java/MCD_Por_Fuerza_Bruta.java
package MCD;

import java.util.Scanner;

public class MCD_Por_Fuerza_Bruta {

    public void main(String[] args) {
        System.out.printf("Ingrese dos números: ");
        Scanner reader = new Scanner(System.in);
        int a = reader.nextInt();
        int b = reader.nextInt();
        if (a > 0 && b > 0) {
            System.out.printf("El MCD(%d,%d)=%d", a, b, MCD(a, b));
        } else {
            System.out.printf("Números inválidos");
        }
    }

    int MCD(int a, int b) {
        int divisor;
        if (a >= b) {
            divisor = b;
        } else {
            divisor = a;
        }
        while (divisor >= 1) {
            if (a % divisor == 0 && b % divisor == 0) {
                return divisor;
            }
            divisor--;
        }
        return 1;
    }
}
